apps/src/gamelab/GameLab.js

```diff
@@ -325,6 +334,7 @@ GameLab.prototype.init = function (config) {
       <GameLabView
         showFinishButton={finishButtonFirstLine && showFinishButton}
         onMount={onMount}
+        onToggleDebugSprites={onToggleDebugSprites}
```

Comment:
There's an antipattern here - a major benefit of putting this property in Redux is that React components can use it and modify it without having to pass a callback down through several layers of view.

Instead what I'd do:

1. In `setupReduxSubscribers()` I'd have `GameLab` watch for `runState.isDebuggingSprites` to change, and tell `this.gameLabP5` about it.
2. JSDebugger.jsx should get `onToggleDebugSprites` from its `connect()` function, as it does with `setStepSpeed` or `addWatchExpression`.

GameLabView.jsx and CodeWorkspace.jsx shouldn't have to change at all for this feature.

Reply:
Makes sense, but how would you suggest that I signal to codeworkspace / jsdebugger that we are in Gamelab and want that button to appear? As currently implemented, the presence of that property (the handler) means that we do want the button to appear when running. Perhaps another Boolean property like a page constant?

Reply:
Makes sense, but how would you suggest that I signal to codeworkspace / jsdebugger that we are in Gamelab and want that button to appear? As currently implemented, the presence of that property (the handler) means that we do want the button to appear when running. Perhaps another Boolean property like a page constant?

Reply:
That's a good point.  `state.pageConstants.appType` should already be available and set to `"gamelab"` - that's probably the easiest way to decide to show this gamelab-specific feature in the debugger.

There might be a cleaner architecture for that (a special gamelab jsdebugger, or injecting a set of buttons into the debugger?) but I'd get it working this way first.